### ISSUE TITLE
Clipboard voicing with a filter for names please?

### ISSUE THREAD

**Issue by Uzukii** [renpy/renpy#4484]
I always use a text-to-speech app when I play Ren'py games. If Shift+C could just copy what's in the text box and not the names too, that would be perfection. Because it's very grating and distracting to have to hear a name said over and over before each line of dialog.

**Comment by renpytom** [comment]
I'm thinking that we probably need to be a bit clever about this - I think it would be impossible to use if it didn't say the name when the character speaking changes.

**Comment by Uzukii** [comment]
I have a method that partially works already, due to how I play my games. And by partial I mean it'll still pronounce the first few letters of a name (or say the whole name if it's short) then overlap/interrupt itself by proceeding with the dialog. It's still frustrating, but close to a remedy I think.

I use two PCs and three essential apps when I play text based games. The apps are Barrier (for sharing the mouse and keyboard), Textractor (for copying the text from games to the clipboard) and Ditto (for sharing the clipboard between the two PCs). One PC plays the game while the other translates, logs or voices the text that was sent to it. It's possible to do everything on a single PC, but it's much, much more cumbersome that way. Textractor works with most 32-bit Ren'py games, so enabling clipboard voicing isn't necessary. It's the 64-bit games that are the problem. They're the ones I have to use the built in clipboard voicing feature with and thus have to hear the names all the time. But, since I'm using two PCs that share clipboards, I can set filters on what gets sent via Textractor by hooking it to Ditto. It's hard to explain, but basically it goes like this:

Game > Text-to-Clipboard > Clipboard-to-Ditto 1 > Ditto 1 to Ditto 2 (other PC) > Ditto 2 to Textractor > Textractor filters names, re-copies text to clipboard > Text-to-Speech app reads it.

The Text-to-Speech app can be used on either PC, but somewhere in between all those steps, the unfiltered clipboard still gets read before the recopy applies. So, before seeking a solution here, I made a thread on Ditto's github about it.
https://github.com/sabrogden/Ditto/issues/451

TL;DR - What we need is a way to use (Regex?) filters on the clipboard so that it ignores certain text phrases. Which in a Ren'py game's case would be any text proceeding a colon mark (:).

**Comment by renpytom** [comment]
Is this an accessibility thing? 

I guess I don't know how if we filter names, it'll be possible to figure out who's speaking.

**Comment by Uzukii** [comment]
No, it's not an accessibility thing, it's a convenience and comfort thing. Having the dialog read for me in a TTS voice I like (Cortana) is better than reading it myself. Avoids eyestrain and makes the visual novels feel like interactive movies when you can use auto-forward. Which you can't when clipboard voicing is enabled... But the names don't have to be read by TTS for me to know who's talking, because I can, and do, still read it.
